services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_service_specs/settings.py

```diff
@@ -198,7 +196,7 @@ def remap_to_compose_spec_key() -> dict[str, str]:
         # remaps from image_name as key to compose_spec key
         return {reverse_mapping[k]: v for k, v in docker_image_name_by_services.items()}
 
-    compose_spec: Optional[ComposeSpecLabel] = cast(
+    compose_spec: ComposeSpecLabel | None = cast(
```

Comment:
is this cast really needed?

Reply:
Yes, the `Json` type there is used to automatically decode the string to an object. This is still required.

Reply:
So apparently if you cast None to an object you can do it and it is still None.
```
from typing import cast


class A:
    pass


none_cast = cast(A, None)
print(none_cast)
assert cast(A, None) is None
```